flake.nix

```diff
@@ -0,0 +1,160 @@
+{
+  inputs = {
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    naersk.url = "github:nix-community/naersk";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = { self, fenix, flake-utils, nixpkgs, ... }@inputs:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = (import nixpkgs) {
+          inherit system;
+        };
+
+        fenix' = fenix.packages.${system};
+
+        toolchainSpec = {
+          channel = "nightly";
+          date = "2022-12-22";
+          sha256 = "sha256-FK01QQuXkFXuy/W7wzAA0G+T2s9dQIDBjMxMC0cUk2M=";
+        };
+
+        platformToTarget = x: {
+          "aarch64-darwin" = "aarch64-apple-darwin";
+          "aarch64-linux" = "aarch64-unknown-linux-gnu";
+          "i686-linux" = "i686-unknown-linux-gnu";
+          "x86_64-darwin" = "x86_64-apple-darwin";
+          "x86_64-linux" = "x86_64-unknown-linux-gnu";
+        }.${x};
```

Comment:
To anyone who's not familiar with `nix` the language. This is a lambda which accepts `x` and returns an element of a dictionary that corresponds to `${x}`, i.e. the string interpolated version. 

While this might seem quaint to an outside observer, the reason why this approach works, is because the function can do any arbitrary transformation. For example, enforce compilation of `i686-linux` for `x86_64` platforms.